src/main/java/com/prgrms/springbootbasic/service/VoucherService.java

```diff
@@ -0,0 +1,34 @@
+package com.prgrms.springbootbasic.service;
+
+import com.prgrms.springbootbasic.domain.FixedDiscountVoucher;
+import com.prgrms.springbootbasic.domain.PercentDiscountVoucher;
+import com.prgrms.springbootbasic.domain.Voucher;
+import com.prgrms.springbootbasic.enums.VoucherType;
+import com.prgrms.springbootbasic.repository.VoucherRepository;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.stereotype.Service;
+
+@Service
+public class VoucherService {
+
+    private final VoucherRepository voucherRepository;
+
+    public VoucherService(VoucherRepository voucherRepository) {
+        this.voucherRepository = voucherRepository;
+    }
+
+    public Voucher createVoucher(VoucherType type, long discount) {
+        Voucher voucher;
+        switch (type) {
+            case FIXED -> voucher = new FixedDiscountVoucher(UUID.randomUUID(), discount);
+            case PERCENT -> voucher = new PercentDiscountVoucher(UUID.randomUUID(), discount);
+            default -> throw new IllegalStateException("입력가능한 Voucher Type은 " + type + "입니다.");
+        }
```

Comment:
switch 표현식은 값을 리턴할 수 있어요!

```suggestion
        Voucher voucher = switch (type) {
            case FIXED -> new FixedDiscountVoucher(UUID.randomUUID(), discount);
            case PERCENT -> new PercentDiscountVoucher(UUID.randomUUID(), discount);
        };
```

Reply:
넵! switch 표현식을 이용해 값을 리턴하도록 변경하였습니다!
해당 커밋: 74aff78